core/src/test/java/io/grpc/util/GracefulSwitchLoadBalancerTest.java

```diff
@@ -0,0 +1,481 @@
+    LoadBalancer lb0 = balancers.get(lbPolicies[0]);
+    Helper helper0 = helpers.get(lb0);
+    SubchannelPicker picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(READY, picker);
+
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[1]));
+    verify(lb0, never()).shutdown();
+
+    LoadBalancer lb1 = balancers.get(lbPolicies[1]);
+    Helper helper1 = helpers.get(lb1);
+    SubchannelPicker picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(CONNECTING, picker1);
+    verify(mockHelper, never()).updateBalancingState(CONNECTING, picker1);
+
+    picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(CONNECTING, picker);
+    verify(lb0).shutdown();
+    verify(mockHelper, never()).updateBalancingState(CONNECTING, picker);
+    verify(mockHelper).updateBalancingState(CONNECTING, picker1);
+
+    picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(READY, picker1);
+    verify(mockHelper).updateBalancingState(READY, picker1);
+
+    verifyNoMoreInteractions(lb0, lb1);
+  }
+
+  @Test
+  public void switchWhileOldPolicyGoesFromReadyToNotReadyWhileNewPolicyStillIdle() {
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[0]));
+    LoadBalancer lb0 = balancers.get(lbPolicies[0]);
+    InOrder inOrder = inOrder(lb0, mockHelper);
+    Helper helper0 = helpers.get(lb0);
+    SubchannelPicker picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(READY, picker);
+
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[1]));
+    inOrder.verify(lb0, never()).shutdown();
+
+    LoadBalancer lb1 = balancers.get(lbPolicies[1]);
+    Helper helper1 = helpers.get(lb1);
+
+    picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(CONNECTING, picker);
+
+    inOrder.verify(lb0).shutdown();
+    inOrder.verify(mockHelper, never()).updateBalancingState(CONNECTING, picker);
+    inOrder.verify(mockHelper).updateBalancingState(IDLE, BUFFER_PICKER);
+
+    picker = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(CONNECTING, picker);
+    inOrder.verify(mockHelper).updateBalancingState(CONNECTING, picker);
+
+    inOrder.verifyNoMoreInteractions();
+    verifyNoMoreInteractions(lb1);
+  }
+
+  @Test
+  public void newPolicyNameTheSameAsPendingPolicy() {
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[0]));
+    LoadBalancer lb0 = balancers.get(lbPolicies[0]);
+    Helper helper0 = helpers.get(lb0);
+    SubchannelPicker picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(READY, picker);
+
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[1]));
+    LoadBalancer lb1 = balancers.get(lbPolicies[1]);
+
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[1]));
+    assertThat(balancers.get(lbPolicies[1])).isSameInstanceAs(lb1);
+
+    verifyNoMoreInteractions(lb0, lb1);
+  }
+
+  @Test
+  public void newPolicyNameTheSameAsCurrentPolicy() {
```

Comment:
Test method names should include both the premises and the expected outcome. For example: "switchToSamePolicyLb_hasNoEffect()".

Reply:
Done. However, did not apply to all methods, some have multiple verification points. Normally I use _suffix for dimensions, such as testMethod1_option1On_option2Off.